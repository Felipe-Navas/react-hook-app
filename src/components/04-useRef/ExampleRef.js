import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

import './effects.css';

export const ExampleRef = () => {
  const [show, setShow] = useState(false);

  const handleHide = () => {
    setShow(!show);
  };

  return (
    <div>
      <h1>Emaple useRef</h1>
      <hr />
      {show && <MultipleCustomHooks />}

      <button className="btn btn-primary mt-4" onClick={handleHide}>
        Show/Hide
      </button>
    </div>
  );
};
