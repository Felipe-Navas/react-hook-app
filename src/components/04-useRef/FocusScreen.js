import React, { useRef } from 'react';
import './effects.css';

export const FocusScreen = () => {
  const inputRef = useRef();

  const handleClick = () => {
    // document.querySelector('input').focus();
    // document.querySelector('input').select();
    inputRef.current.select();
  };

  return (
    <div>
      <h1>FocusScreen</h1>
      <hr />

      <input
        ref={inputRef}
        className="form-control"
        placeholder="Su nombre"
        type="text"
      />

      <button className="mt-2 btn btn-success" onClick={handleClick}>
        Focus
      </button>
    </div>
  );
};
