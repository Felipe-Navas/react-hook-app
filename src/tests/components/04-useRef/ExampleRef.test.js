import React from 'react';
import { shallow } from 'enzyme';
import { ExampleRef } from '../../../components/04-useRef/ExampleRef';

describe('Testing the <ExampleRef /> component', () => {
  const wrapper = shallow(<ExampleRef />);
  test('should render', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });

  test('should render the component <MultipleCustomHooks /> when click the button', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });
});
