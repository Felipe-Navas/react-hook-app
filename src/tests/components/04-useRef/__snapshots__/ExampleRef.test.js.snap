// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <ExampleRef /> component should render 1`] = `
<div>
  <h1>
    Emaple useRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
